packages/toolpad-app/src/toolpad/Home/index.tsx

```diff
@@ -124,6 +125,11 @@ function CreateAppDialog({ onClose, ...props }: CreateAppDialogProps) {
     },
   });
 
+  const [latestStoredApp] = useLocalStorageState<LatestStoredAppValue>(
```

Comment:
This hook has proven to be so useful already. The only downside I see at the moment is that the type may change from one version to the next. typescript won't help here to prevent or detect that. I'm thinking of adding serialization options with `superstruct` to guard what's stored and provide the types

Reply:
Would be really cool to be able to pass in a key and get the correct expected types, it would probably require some thinking on how to implement it though.

Reply:
I roughly had someting like this in mind
```tsx
import { object, number, string } from 'superstruct'
import useLocalStorageState from './utils/useLocalStorageState';

const LastCreatedApp = object({
  id: string(),
  lastSavedAt: number()
})

// ...

  const [lastCreatedApp, setLastCreatedApp] = useLocalStorageState('my-article', null, LastCreatedApp)
```

Reply:
Could be, or might be a bit heavy to pass in a schema just for typing, especially if we use the hook in multiple places...

Maybe we could just have an interface / type object with type definitions for keys and values and pass that to useLocalStorageState as a type parameter? And maybe an abstraction called `useToolpadLocalStorageState` that already included those types? Or that might be too much abstraction... Just some rough ideas anyway.

I guess we could discuss this later as a separate issue.

Reply:
It's not just for typing, it needs to actually validate in runtime that whatever came from localstorage is compatible with with that schema